static/src/components/ScheduleModal.js

```diff
@@ -0,0 +1,197 @@
+    portion: Yup.number().required().min(0.0625).max(2),
+    enabled: Yup.bool().required()
+});
+
+const twentyFourHourToSeconds = (hour, minute) => {
+    const totalMin = (hour * 60) + minute
+    return totalMin * 60
+}
+
+const secondsToTwentyFourHour = (second) => {
+    const hour = Math.floor(second / 3600)
+    const minute = (second % 3600) / 60
+    return {hour, minute}
+}
+
+
+export const ScheduleModalComponent = function (props) {
+    let scheduledEvents = <Card bg={"primary"} className={"text-light text-center mb-2"}>
+        <Card.Body>
+            No Meals Scheduled
+        </Card.Body>
+    </Card>
+    if (props.hasOwnProperty("events") && props.events && props.events.length > 0) {
+        scheduledEvents = props.events.map((event) =>
+            <Card bg={event.enabled ? "primary" : "info"} className={"text-light mb-3"}>
+                <Card.Body>
+                    <Row>
+                        <Col xs={12} sm={9}>
+                            <h3>{event.name}</h3>
+                            <p className={'mb-0 mr-3'} style={{display: 'inline-block'}}>
+                                <Icon path={mdiCup}
+                                      size={.75}/> {displaySizes[event.portion]} {event.portion > 1 ? "cups" : "cup"}
+                            </p>
+                            <p className={'mb-3'} style={{display: 'inline-block'}}>
+                                <Icon path={mdiClockOutline} size={.75}/> {secSinceMidnightFormatter(event.time)}
+                            </p>
+                        </Col>
+                        <Col xs={12} sm={3}>
+                            <Button size="sm" variant={"light"} className={"mt-2 mb-1"} style={{width: '100%'}}
+                                    onClick={() => props.startEdit(false, event)}>
+                                <Icon path={mdiSquareEditOutline} size={.75}/> Edit
+                            </Button>
+                            <Button size="sm" variant={"danger"} className={"mt-1 mb-0"}
+                                    style={{width: '100%'}} onClick={() => props.handleDeleteEvent(event.event_id)}>
+                                <Icon path={mdiDelete} size={.75}/> Delete
+                            </Button>
+                        </Col>
+                    </Row>
+                </Card.Body>
+            </Card>)
+    }
+
+    let body = <>
+        <Modal.Body>
+            {scheduledEvents}
+            <Button
+                onClick={() => props.startEdit(true)}
+                variant={'secondary'}
+                style={{width: '100%'}}
+                className={'mt-1'}
+            >
+                Add Meal
+            </Button>
+        </Modal.Body>
+    </>
+    if (props.editMode) {
+        body = <Formik
+            initialValues={Object.keys(props.targetEvent).length !== 0 ? props.targetEvent : {
+                name: "",
+                time: 39600,
```

Comment:
Is this 11AM?

Reply:
Yah, I just kinda picked something. I was going to do the current time, but that seemed like extra complexity for no real improvement in UX.